File: Tracks/typescript/isbn-verifier/isbn-verifier.ts
export default class ISBN {
  private readonly isbn: string;
  constructor(isbn: string) {
    this.isbn = isbn.replace(/-/g, "");
  }

  isValid = (): boolean => {
    if (this.isbn.length !== 10) return false;
    if (this.isbn.match(/[0-9]{9}[0-9X]{1}/g) === null) return false;
    return (
      [...this.isbn].reduce(
        (total, current, index) =>
          total + (current === "X" ? 10 : parseInt(current)) * (10 - index),
        0
      ) % 11 === 0
    );
  };
}

File: Tracks/csharp/isbn-verifier/IsbnVerifier.cs
using System.Linq;
using System.Text.RegularExpressions;

public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        var match = new Regex(@"^\d{1}-?\d{3}-?\d{5}-?[\dX]{1}$").Match(number);
        if (!match.Success) return false;

        static (int number, int index) ConvertToDigit(char digitOrX, int index) => (digitOrX == 'X' ? 10 : int.Parse("" + digitOrX), index);
        static int AggregateNumbers(int previous, (int number, int index) current) => previous + current.number * (10 - current.index);

        return match.Value.Where(char.IsLetterOrDigit).Select(ConvertToDigit).Aggregate(0, AggregateNumbers) % 11 == 0;
    }
}